jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/AclAuthorizerBenchmark.java

```diff
@@ -164,4 +206,28 @@ public void testAclsIterator() {
     public void testAuthorizer() {
         aclAuthorizer.authorize(context, actions);
     }
+
+    @Benchmark
+    public void testAuthorizeByResourceType() {
+        aclAuthorizer.authorizeByResourceType(context, AclOperation.WRITE, ResourceType.TOPIC);
+    }
+
+    @Benchmark
+    public void testUpdateCache() {
```

Comment:
If you take an async profile of this benchmark method you end up spending most of the time in building the entries and immutable set, and barely any time on `AclAuthorizer#updateCache`.

Reply:
Yeah. Agree. Let's see what @rajinisivaram think about the above discussion